src/main/java/webserver/Router.java

```diff
@@ -10,19 +10,28 @@
 import webserver.request.Request;
 import webserver.request.HTTP;;
 
-
 public class Router {
     private HashMap<String, Callback<Request, String>> GETRoutes = new HashMap<>();
+    private HashMap<String, Callback<Request, String>> HEADRoutes = new HashMap<>();
```

Comment:
Here's a place where it's generally recommended to use a more generic interface as the declared type (and why there's all that repetition in declaring types for both the field and its value).

"Require no more (than you need); promise no less (than you will deliver)" is the general advice.

http://www.javapractices.com/topic/TopicAction.do?Id=194


TL;DR saying `Map<String, Callback...> headRoutes = new HashMap<>();` allows you to eventually accept types like `TreeMap` or `Hashtable` if those specific implementations are needed.

Reply:
Oh gotcha! Thanks!